spec.md

```diff
@@ -325,19 +330,28 @@ on the definition of OPTIONAL.
   event format defines the relationship in
   [section 3.1](./json-format.md#31-handling-of-data).
 
-  When this attribute is omitted, `data` simply follows the event
-  format's encoding rules. For the JSON event format, the `data` value
-  can therefore be a JSON object, array, or value.
-
-  For the binary mode of some of the CloudEvents transport bindings, where the
-  `data` content is immediately mapped into the payload of the transport frame,
-  this field is directly mapped to the respective transport or application
+  For the binary mode of some of the CloudEvents protocol bindings, where the
+  `data` content is immediately mapped into the payload of the protocol frame,
+  this field is directly mapped to the respective protocol or application
   protocol's content-type metadata property. Normative rules for the binary mode
-  and the content-type metadata mapping can be found in the respective transport
+  and the content-type metadata mapping can be found in the respective protocol
   mapping specifications.
 
+  This attribute MAY be omitted if the event format has an implied
+  `datacontenttype`. For example, the JSON format implies "application/json". A
+  JSON-format event with no `datacontenttype` is exactly equivalent to one with
+  `datacontenttype="application/json"` and its `data` can be any JSON object,
+  array, or value.
+
+  Formats and protocols that do not have an implied `datacontenttype` MUST NOT
```

Comment:
I believe this is really just talking about binary mode, right?
If so, I have mixed feeling about this. While I do agree that it's a good idea, best practice, to include a content-type on all messages, I tend to think of binary mode being used in cases where CE is being added to an existing message (while not always true, I do think that's going to be a common case). And so in those cases we have an existing message flow that works w/o a content-type attribute, and now the addition of CE will mandate that senders include an attribute that previously wasn't needed. That seems like an unnecessary burden and out of scope for CE to mandate since we're not trying to control how the payload of the message is processed - just how to find common metadata.

Now.... on the flip side, as I said, I do think it's a good idea to include it and I can certainly understand someone thinking that CE isn't just about CE attributes but also about promoting interop and so nudging people to include a content-type attribute is just an overall good thing.

Would it be horrible if instead of making it a MUST for binary mode, we make it a SHOULD or STRONGLY RECOMMENDED ?

Reply:
Lemme take another shot.